src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -357,11 +509,63 @@ void TileEmitter::emit_body(const std::vector<size_t>& vec_pool, const std::vect
-    for (size_t i = 0; i < num_inputs + num_outputs; i++) {
-        // those with dims == 1 will be broadcasted, hence don't require increment
-        if (io_dims[i] != 1)
-            h->add(data_ptr_regs[i], increment * io_data_size[i]);
+void TileEmitter::emit_ptr_increments_static(const std::vector<Reg64>& data_ptr_regs) const {
+    auto master_shape_last_dim = *std::max_element(io_dims.begin(), io_dims.end());
+    for (const auto& idx : static_dims_idx) {
+        // increment only inputs that are not broadcasted
+        if (io_dims[idx] != 1 || master_shape_last_dim == 1)
+            h->add(data_ptr_regs[idx], increment * io_data_size[idx]);
+    }
+}
+
+void TileEmitter::emit_ptr_increments_dynamic(const Reg64& reg_const_params, const std::vector<Reg64>& data_ptr_regs) const {
+    emit_ptr_increments_static(data_ptr_regs);
+    const size_t tile_type_offset = increment > 1 ? GET_OFF(vector_tile_increments) : GET_OFF(scalar_tile_increments);
+    for (size_t i = 0; i < dynamic_dims_idx.size(); i++) {
+        auto idx = dynamic_dims_idx[i];
+        h->add(data_ptr_regs[idx], h->ptr[reg_const_params + tile_type_offset + idx * sizeof(int64_t)]);
+    }
+}
+
+template <dnnl::impl::cpu::x64::cpu_isa_t isa>
+void TileEmitter::set_increments_and_broadcast_inputs(const Reg64& reg_const_params, const std::vector<Reg64> &data_ptr_regs) const {
+        using Vmm = typename dnnl::impl::utils::conditional3<isa == dnnl::impl::cpu::x64::sse41,
+                                                             Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
+        auto Vmm_tmp = Vmm(0);
+        for (size_t i = 0; i < dynamic_dims_idx.size(); i++) {
+            auto idx = dynamic_dims_idx[i];
+            const auto& data_ptr_reg = data_ptr_regs[idx];
+            // todo: we can store dynamic broadcasting info only for dynamic inputs (not for all, like we do now)
+            h->cmp(h->byte[reg_const_params + GET_OFF(broadcasting_mask) + idx * sizeof(bool)], 0);
+            Label no_broadcasting;
+            h->je(no_broadcasting, CodeGenerator::T_SHORT);
+            // Both inputs and outputs can be dynamic, but only inputs could be physically broadcasted
+            // Physical broadcasting is only required for vector tiles
+            if (idx < num_inputs && increment != 1) {
+                h->push(data_ptr_reg);
+                h->uni_vbroadcastss(Vmm_tmp, h->ptr[data_ptr_reg]);
+                h->mov(data_ptr_reg, h->ptr[reg_const_params + GET_OFF(broadcasting_scratchpad)]);
+                h->add(data_ptr_reg, i * increment *  io_data_size[idx]);
+                // note that we use data_ptr_reg directly without h->rip
+                h->uni_vmovups(h->ptr[data_ptr_reg], Vmm_tmp);
+            }
+            h->L(no_broadcasting);
+        }
+}
+
+void TileEmitter::cleanup_broadcasting(const Reg64& reg_const_params, const std::vector<Reg64> &data_ptr_regs) const {
+    if (increment == 1)
```

Comment:
This is done intentionally to align calls of  `set_increments_and_broadcast_inputs` and `cleanup_broadcasting`. Otherwise it might be confusing why do we call `set_increments_and_broadcast_inputs` for all tiles, but cleanup_broadcasting` only for scalar ones. An alternative solution is to split `set_increments_and_broadcast_inputs` into `set_increments` and `broadcast_inputs` and call the latter one only for vector tiles, but this would be less efficient, since it would require a second loop over dynamic_dims_idx